### PR TITLE
chore: Document how to use form controls in a Field (Set)

### PR DESCRIPTION
**Pull request by alimpens**
# Describe the pull request

Thank you for contributing to the project!
Please use this template to help us handle your PR smoothly.

## What

It adds 2 stories to all input docs, showing how you wrap the input in a Field (Set) and how you can show that the Field (Set) is invalid. 

## Why

We only explained how Field and Field Set work on those pages, not on the input pages. This makes those explanations harder to miss. 

## How

By adding 2 stories and some docs to all input pages. 

## Checklist

Before submitting your pull request, please ensure you have done the following. Check each checkmark if you have done so or if it wasn't necessary:

- [x] Add or update unit tests
- [x] Add or update documentation
- [x] Add or update stories
- [x] Add or update exports in index.\* files
- [x] Start the PR title with a Conventional Commit prefix, [as explained here](https://github.com/Amsterdam/design-system/blob/main/documentation/publishing.md?plain=1#L11).

## Additional notes

- Provide links to any related issues or discussions.
- Add a link to the specific story in the feature branch deploy.
- Mention any areas where additional review or feedback is needed.

### CODE REVIEW

File: storybook/src/components/Radio/Radio.docs.mdx [hunk 1]
```diff
@@ -12,7 +12,27 @@ import README from "../../../../packages/css/src/components/radio/README.md?raw"
...
+
+If you don’t need the description, just take out the description Paragraph and the `aria-describedby` from the Field Set.
+
+For more information, check [the Field Set component documentation](/docs/components-forms-field-set--docs).
+
+<Canvas of={RadioStories.InAFieldSet} />
+
+### In an invalid Field Set
+
+This is an example of an invalid Field Set with a Label, a description, an Error Message and a group of Radio’s.
```
Comment (VincentSmedinga) [inline storybook/src/components/Radio/Radio.docs.mdx:34]:
```suggestion
If the Radio can become invalid, add an Error Message component and its `id` to the `aria-describedby` prop of the Field Set.
```

File: storybook/src/components/Radio/Radio.docs.mdx [hunk 2]
```diff
@@ -12,7 +12,27 @@ import README from "../../../../packages/css/src/components/radio/README.md?raw"
...
+Usually, you should place Radio’s inside a Field Set component.
+Here’s an example of a Field Set component that includes a legend, a description, and a group of Radio’s.
+
+When grouping radio inputs, use `role="radiogroup"` on Field Set to have this grouping explicitly announced as a radio group (the default role is group).
+Using `role="radiogroup"` also allows you to use `aria-required` on Field Set, which isn’t allowed for role group.
+Always also set `aria-required` on the individual radio buttons though, to make sure it’s read by screen readers.
+
+If you don’t need the description, just take out the description Paragraph and the `aria-describedby` from the Field Set.
+
+For more information, check [the Field Set component documentation](/docs/components-forms-field-set--docs).
```
Comment (VincentSmedinga) [inline storybook/src/components/Radio/Radio.docs.mdx:28]:
```suggestion
Check [Field Set](/docs/components-forms-field-set--docs) for more information on configuring it.
```

File: storybook/src/components/Radio/Radio.docs.mdx [hunk 3]
```diff
@@ -12,7 +12,27 @@ import README from "../../../../packages/css/src/components/radio/README.md?raw"
...
 
-Group radios together with a [Field Set](/docs/components-forms-field-set--docs) that describes them.
-This is usually a question, like ‘Where do you live?’.
+## Examples
 
-<Canvas of={RadioStories.RadioGroup} />
+### In a Field Set
+
+Usually, you should place Radio’s inside a Field Set component.
+Here’s an example of a Field Set component that includes a legend, a description, and a group of Radio’s.
```
Comment (VincentSmedinga) [inline storybook/src/components/Radio/Radio.docs.mdx:20]: This is more to the point – is it strong enough to combine these two lines?

> Use a Field Set to group several Radios with a label, description, and error message.

File: storybook/src/components/Radio/Radio.stories.tsx [hunk 4]
```diff
@@ -64,33 +64,92 @@ export const RadioGroup: Story = {
...
+    },
+    children: {
+      table: { disable: true },
+    },
+    disabled: {
+      table: { disable: true },
+    },
+    onChange: {
+      action: 'clicked',
+      table: { disable: true },
```
Comment (VincentSmedinga) [inline storybook/src/components/Radio/Radio.stories.tsx:121]: Ha! :) Didn’t you ask me to revert those when I added them lately? Or did we choose hiding unfocused args as the way to go? I don’t remember which PR it was exactly and I can’t find it when browsing a couple.

File: storybook/src/components/Radio/Radio.stories.tsx [hunk 5]
```diff
@@ -64,33 +64,92 @@ export const RadioGroup: Story = {
...
     },
   },
+  render: (args) => (
+    <FieldSet
+      aria-describedby={`description2${args.invalid ? ' error2' : ''}`}
```
Comment (VincentSmedinga) [inline storybook/src/components/Radio/Radio.stories.tsx:126]: This isn’t the improvement I was hoping for, is it?

```suggestion
      aria-describedby={['description2', args.invalid ? 'error2' : undefined].join('')}`}
```

File: storybook/src/components/Radio/Radio.stories.tsx (hunk 4 above)
Comment (alimpens) [reply storybook/src/components/Radio/Radio.stories.tsx:121]: I think I remember what you mean, you only wanted to show the relevant control for a story. So for the 'disabled' story you would only show the 'disabled' control, I think. I (still) prefer showing all controls, so you can easily see what disabled checked with a different label looks like, for example.

The situation is a little different here. This story has a collection of Radios, so controls like `checked` and `disabled` don't really have a obvious thing they control. Is it the first Radio, all the Radios, something else?

For that reason, I removed them altogether. We could choose a different approach though.

File: storybook/src/components/Radio/Radio.stories.tsx (hunk 5 above)
Comment (alimpens) [reply storybook/src/components/Radio/Radio.stories.tsx:126]: It's not much clearer imo, and the current code doesn't work :) You could join with `' '`, but that means you get this when `invalid` is false: `"description2 "`

File: storybook/src/components/FieldSet/FieldSet.docs.mdx [hunk 6]
```diff
@@ -36,24 +36,25 @@ Add an `aria-describedby` attribute to the input and provide the `id` of Error M
 
 ### Radio group
 
-Use a Field Set to group radio buttons.
-When grouping radio inputs, use `role="radiogroup"` on Field Set to have this grouping explicitly announced as a radio group (the default role is `group`).
-Using `role="radiogroup"` also allows you to use `aria-required` on Field Set, which isn’t allowed for role `group`.
-Always also set `aria-required` on the individual radio buttons though, to make sure it’s read by screen readers.
+Use a Field Set to group several Radios with a legend, description, and / or an error message.
+
+When grouping Radios, add `role="radiogroup"` to the Field Set to have it explicitly announced as a radio group (the default role is just ‘group’).
```
Comment (VincentSmedinga) [inline storybook/src/components/FieldSet/FieldSet.docs.mdx:41]: No need to repeat the grouping once more.

```suggestion
Add `role="radiogroup"` to the Field Set to have it explicitly announced as a radio group (the default role is just ‘group’).
```

File: storybook/src/components/FieldSet/FieldSet.docs.mdx [hunk 7]
```diff
@@ -36,24 +36,25 @@ Add an `aria-describedby` attribute to the input and provide the `id` of Error M
...
+The ‘radio group’ role also allows using `aria-required` on Field Set; that isn’t allowed for the ‘group’ role.
+Always also set `aria-required` on the individual Radios though, to make sure it’s read by screen readers.
```
Comment (VincentSmedinga) [inline storybook/src/components/FieldSet/FieldSet.docs.mdx:43]: ‘Always also’ and ‘though’ feel a bit off. Grammarly suggests:

```suggestion
However, always set `aria-required` on the individual Radios to ensure that screen readers can read it.
```

We could replace ‘However’ with ‘Still’ or ‘Even so’.

File: storybook/src/components/FieldSet/FieldSet.docs.mdx [hunk 8]
```diff
@@ -36,24 +36,25 @@ Add an `aria-describedby` attribute to the input and provide the `id` of Error M
...
 
 <Canvas of={FieldSetStories.RadioGroup} />
 
 ### Radio group with error
```
Comment (VincentSmedinga) [inline storybook/src/components/FieldSet/FieldSet.docs.mdx:47]: Should we also say ‘With Validation’ here, as we do for the Form components?

File: storybook/src/components/Radio/Radio.docs.mdx [hunk 9]
```diff
@@ -12,7 +12,26 @@ import README from "../../../../packages/css/src/components/radio/README.md?raw"
...
-Group radios together with a [Field Set](/docs/components-forms-field-set--docs) that describes them.
-This is usually a question, like ‘Where do you live?’.
+## Examples
 
-<Canvas of={RadioStories.RadioGroup} />
+### In a Field Set
+
+Use a Field Set to group several Radios with a legend, description and / or an error message.
+
+When grouping Radios, add `role="radiogroup"` to the Field Set to have it explicitly announced as a radio group (the default role is just ‘group’).
```
Comment (VincentSmedinga) [inline storybook/src/components/Radio/Radio.docs.mdx:21]: No need to repeat the grouping once more.

```suggestion
Add `role="radiogroup"` to the Field Set to have it explicitly announced as a radio group (the default role is just ‘group’).
```

File: storybook/src/components/TextArea/TextArea.docs.mdx [hunk 10]
```diff
@@ -31,3 +31,21 @@ import README from "../../../../packages/css/src/components/text-area/README.md?
 ## Disabled
 
 <Canvas of={TextAreaStories.Disabled} />
+
+### In a Field
+
+Use a Field component to group a Text Area with a Label, description and / or an error message.
+
```
Comment (VincentSmedinga) [inline storybook/src/components/TextArea/TextArea.docs.mdx:38]:
```suggestion
Use a Field to group a Text Area with a Label, description and / or an Error Message.

```

File: storybook/src/components/TextArea/TextArea.docs.mdx [hunk 11]
```diff
@@ -31,3 +31,21 @@ import README from "../../../../packages/css/src/components/text-area/README.md?
...
+If you don’t need the description, just take out the description Paragraph and the `aria-describedby` from the Text Input.
+
+Check [the Field component documentation](/docs/components-forms-field--docs) for more information on configuring it.
```
Comment (VincentSmedinga) [inline storybook/src/components/TextArea/TextArea.docs.mdx:41]:
```suggestion
Check [Field](/docs/components-forms-field--docs) for more information on configuring it.
```

Or ‘Consult [the Field docs]` if you really want the ‘documentation’ piece in?

File: storybook/src/components/TextArea/TextArea.docs.mdx [hunk 12]
```diff
@@ -31,3 +31,21 @@ import README from "../../../../packages/css/src/components/text-area/README.md?
...
+
+<Canvas of={TextAreaStories.InAField} />
+
+### In a Field With Validation
+
+If the Text Area can become invalid, add an Error Message component and its `id` to the `aria-describedby` prop of the Text Area.
+
+Check [the Field component documentation](/docs/components-forms-field--docs) for more information on configuring it.
```
Comment (VincentSmedinga) [inline storybook/src/components/TextArea/TextArea.docs.mdx:49]:
```suggestion
Check [Field](/docs/components-forms-field--docs) for more information on configuring it.
```

File: storybook/src/components/TextInput/TextInput.docs.mdx [hunk 13]
```diff
@@ -113,6 +113,24 @@ Alternatives:
 
 <Canvas of={TextInputStories.Disabled} />
 
+### In a Field
+
+Use a Field component to group a Text Input with a Label, description and / or an error message.
```
Comment (VincentSmedinga) [inline storybook/src/components/TextInput/TextInput.docs.mdx:118]:
```suggestion
Use a Field to group a Text Input with a Label, description and / or an Error Message.
```

File: storybook/src/components/Radio/Radio.docs.mdx [hunk 14]
```diff
@@ -12,7 +12,26 @@ import README from "../../../../packages/css/src/components/radio/README.md?raw"
...
+The ‘radio group’ role also allows using `aria-required` on Field Set; that isn’t allowed for the ‘group’ role.
+Always also set `aria-required` on the individual Radios though, to make sure it’s read by screen readers.
+
+If you don’t need the description, just take out the description Paragraph and the `aria-describedby` from the Field Set.
```
Comment (VincentSmedinga) [inline storybook/src/components/Radio/Radio.docs.mdx:25]:
```suggestion
If you don’t need the description, remove its Paragraph and the `aria-describedby` from the Field Set.
```

File: storybook/src/components/TextArea/TextArea.docs.mdx [hunk 15]
```diff
@@ -31,3 +31,21 @@ import README from "../../../../packages/css/src/components/text-area/README.md?
 ## Disabled
 
 <Canvas of={TextAreaStories.Disabled} />
+
+### In a Field
+
+Use a Field component to group a Text Area with a Label, description and / or an error message.
+
+If you don’t need the description, just take out the description Paragraph and the `aria-describedby` from the Text Input.
```
Comment (VincentSmedinga) [inline storybook/src/components/TextArea/TextArea.docs.mdx:39]:
```suggestion
If you don’t need the description, remove its Paragraph and the `aria-describedby` from the Text Input.
```

File: storybook/src/components/TextArea/TextArea.docs.mdx [hunk 16]
```diff
@@ -31,3 +31,21 @@ import README from "../../../../packages/css/src/components/text-area/README.md?
...
+
+Check [the Field component documentation](/docs/components-forms-field--docs) for more information on configuring it.
+
+<Canvas of={TextAreaStories.InAField} />
+
+### In a Field With Validation
+
+If the Text Area can become invalid, add an Error Message component and its `id` to the `aria-describedby` prop of the Text Area.
```
Comment (VincentSmedinga) [inline storybook/src/components/TextArea/TextArea.docs.mdx:47]:
```suggestion
If the Text Area can become invalid, add an Error Message and its `id` to the `aria-describedby` prop of the Text Area.
```

File: storybook/src/components/TextInput/TextInput.docs.mdx [hunk 17]
```diff
@@ -113,6 +113,24 @@ Alternatives:
...
+
+If you don’t need the description, just take out the description Paragraph and the `aria-describedby` from the Text Input.
```
Comment (VincentSmedinga) [inline storybook/src/components/TextInput/TextInput.docs.mdx:120]:
```suggestion
If you don’t need the description, remove its Paragraph and the `aria-describedby` from the Text Input.
```

File: storybook/src/components/TextInput/TextInput.docs.mdx [hunk 18]
```diff
@@ -113,6 +113,24 @@ Alternatives:
...
+
+Check [the Field component documentation](/docs/components-forms-field--docs) for more information on configuring it.
```
Comment (VincentSmedinga) [inline storybook/src/components/TextInput/TextInput.docs.mdx:122]:
```suggestion
Check [Field](/docs/components-forms-field--docs) for more information on configuring it.
```

File: storybook/src/components/TextInput/TextInput.docs.mdx [hunk 19]
```diff
@@ -113,6 +113,24 @@ Alternatives:
...
+
+<Canvas of={TextInputStories.InAField} />
+
+### In a Field With Validation
+
+If the Text Input can become invalid, add an Error Message component and its `id` to the `aria-describedby` prop of the Text Input.
```
Comment (VincentSmedinga) [inline storybook/src/components/TextInput/TextInput.docs.mdx:128]:
```suggestion
If the Text Input can become invalid, add an Error Message and its `id` to the `aria-describedby` prop of the Text Input.
```

File: storybook/src/components/TextInput/TextInput.docs.mdx [hunk 20]
```diff
@@ -113,6 +113,24 @@ Alternatives:
...
+
+Check [the Field component documentation](/docs/components-forms-field--docs) for more information on configuring it.
```
Comment (VincentSmedinga) [inline storybook/src/components/TextInput/TextInput.docs.mdx:130]:
```suggestion
Check [Field](/docs/components-forms-field--docs) for more information on configuring it.
```

**Review by VincentSmedinga — COMMENTED** [review]
🤓

File: storybook/src/components/FieldSet/FieldSet.docs.mdx (hunk 7 above)
Comment (alimpens) [reply storybook/src/components/FieldSet/FieldSet.docs.mdx:43]: I like 'Even so', it makes it clearer that you also have to set `aria-required` on the individual Radios

File: storybook/src/components/TextArea/TextArea.docs.mdx (hunk 11 above)
Comment (alimpens) [reply storybook/src/components/TextArea/TextArea.docs.mdx:41]: Just 'Field' is a bit terse imo. How about 'Check the Field docs ...'?

File: storybook/src/components/Radio/Radio.stories.tsx (hunk 4 above)
Comment (VincentSmedinga) [reply storybook/src/components/Radio/Radio.stories.tsx]: I’ve added this to the Storybook guidelines.

File: storybook/src/components/TextArea/TextArea.docs.mdx (hunk 11 above)
Comment (VincentSmedinga) [reply storybook/src/components/TextArea/TextArea.docs.mdx]: Alright. I’ve applied this to all references like this one.

**Review by VincentSmedinga — APPROVED** [review]

**Pull request merged**